conan/tools/cmake/cmake.py

```diff
@@ -67,28 +65,17 @@ def configure(self, variables=None, build_script_folder=None):
             cmakelist_folder = os.path.join(self._conanfile.source_folder, build_script_folder)
 
         build_folder = self._conanfile.build_folder
-        generator_folder = self._conanfile.generators_folder
-
         mkdir(self._conanfile, build_folder)
 
         arg_list = [self._cmake_program]
+        # Running the initial cache file
+        if variables:
+            update_cmake_initial_cache(self._conanfile, variables)
```

Comment:
Maybe this is not necessary. If ``CMake`` helper wants to pass things in the command line, let it do it. At this point it is to late for the user to be able to use the cache file generated by the toolchain, so this can probably be simplified.

Reply:
Maybe, but I only wanted to keep all the possible flags in one file because if the user passes more than one, it'll be helpful to have all of them saved. Anyway, I can change this behavior if you think so